app/models/services/facebook.rb

```diff
@@ -23,16 +23,16 @@ def post_to_facebook(url, body)
   end
 
   def create_post_params(post)
-    message = post.message.plain_text_without_markdown
-    if post.photos.any?
-      message += " " + short_post_url(post, protocol: AppConfig.pod_uri.scheme,
-                                            host: AppConfig.pod_uri.authority)
+    post_message = post.message.plain_text_without_markdown
+
+    if post_message.empty? && post.photos.none?
```

Comment:
A post should never be empty and have no photos, there is a validation for that:

https://github.com/diaspora/diaspora/blob/beeb073dbe0d95b0dd38c91f8b129fc0039bd3a0/app/models/status_message.rb#L20-L21

Reply:
Oh ok, I didn't know that so wanted to cover that case.  I can adjust that along with the below suggestion.